### PR TITLE
✨ Cavatica Models and Endpoints

### PR DESCRIPTION
**Pull request by znatty22**
Addresses #193, #287, #298 

- [x] Implement & test model + /cavatica-tasks
- [x] Implement & test model + /cavatica-apps
- [x] Implement & test model + /cavatica-task-genomic-files
- [x] Update & test model + /genomic-files
- [x] Migrations
- [x] Entity rel diagram

### CODE REVIEW

File: dataservice/api/cavatica_task/models.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    :param kf_id: Unique id given by the Kid's First DCC
+    :param created_at: Time of object creation
+    :param modified_at: Last time of object modification
+    :param name: Name of cavatica_task
+    :param external_cavatica_task_id: Id of executed task assigned by Cavatica
+    """
+    __tablename__ = 'cavatica_task'
+    __prefix__ = 'CT'
+
+    external_cavatica_task_id = db.Column(db.Text(),
```
Comment (dankolbman) [inline dataservice/api/cavatica_task/models.py]: Should be a uuid column?

File: dataservice/api/cavatica_app/schemas.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+from dataservice.api.common.schemas import BaseSchema
+from dataservice.extensions import ma
+from dataservice.api.common.validation import validate_positive_number
+
+
+class CavaticaAppSchema(BaseSchema):
+
+    revision = field_for(CavaticaApp, 'revision',
+                         validate=validate_positive_number)
+
```
Comment (dankolbman) [inline dataservice/api/cavatica_app/schemas.py]: Use a `URLField` for the github url?

**Review by dankolbman — COMMENTED** [review]
How difficult is it to get rid of the relationship endpoint and integrate it into the genomic-files and cavatica-tasks endpoints and responses?

File: tests/cavatica_task_genomic_file/test_cavatica_task_gf_resources.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+                         sequencing_center=sc,
+                         participant=p)
+
+        # SequencingExperiment
+        data = {
+            'external_id': 'se',
+            'experiment_strategy': 'wgs',
+            'is_paired_end': True,
+            'platform': 'platform',
+            'sequencing_center': sc
```
Comment (parimalak) [inline tests/cavatica_task_genomic_file/test_cavatica_task_gf_resources.py:209]: should be 'sequencing_center_id': sc.kf_id

File: tests/test_api.py [hunk 4]
```diff
@@ -172,14 +158,18 @@ def test_unknown_field(self, client, entities, endpoint, method):
...
+        ('/genomic-files', ['cavatica_task_genomic_files'])
     ])
     def test_child_links(self, client, entities, resource, fields):
         """ Checks that references to other resources have correct ID """
+        from pprint import pprint
         kf_id = entities.get('kf_ids').get(resource)
         resp = client.get(resource + '/' + kf_id)
         body = json.loads(resp.data.decode('utf-8'))['results']
-
+        pprint(body)
```
Comment (parimalak) [inline tests/test_api.py]: 💣

File: tests/cavatica_task_genomic_file/test_cavatica_task_gf_resources.py (hunk 3 above)
Comment (znatty22) [reply tests/cavatica_task_genomic_file/test_cavatica_task_gf_resources.py:209]: Its the same thing as using the db relationship (sequencing_center). I don't have the kf_id for the sequencing center at this point, so instead i'm just using the sequencing_center obj

File: tests/cavatica_app/test_cavatica_app_models.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+class ModelTest(FlaskTestCase):
+    """
+    Test CavaticaApp database model
+    """
+
+    def test_create_and_find(self, mock):
+        """
+        Test create cavatica_app
+        """
+        indexd = MockIndexd()
```
Comment (dankolbman) [inline tests/cavatica_app/test_cavatica_app_models.py:32]: You may be able to use `utils.IndexdTestCase` instead

**Review by dankolbman — APPROVED** [review]

**Review by dankolbman — APPROVED** [review]
👍

**Review by parimalak — APPROVED** [review]
👍

**Pull request merged**